cpp/src/io/orc/reader_impl.cu

```diff
@@ -1008,20 +1027,40 @@ std::unique_ptr<column> reader::impl::create_empty_column(const int32_t orc_col_
 // Adds child column buffers to parent column
 column_buffer&& reader::impl::assemble_buffer(const int32_t orc_col_id,
                                               std::vector<std::vector<column_buffer>>& col_buffers,
-                                              const size_t level)
+                                              const size_t level,
+                                              rmm::cuda_stream_view stream)
 {
   auto const col_id = _col_meta.orc_col_map[level][orc_col_id];
   auto& col_buffer  = col_buffers[level][col_id];
 
   col_buffer.name = _metadata->get_column_name(0, orc_col_id);
-  switch (col_buffer.type.id()) {
-    case type_id::LIST:
-    case type_id::STRUCT:
+  auto kind       = _metadata->get_col_type(orc_col_id).kind;
+  switch (kind) {
+    case orc::LIST:
+    case orc::STRUCT:
       for (auto const& col : _metadata->get_col_type(orc_col_id).subtypes) {
-        col_buffer.children.emplace_back(assemble_buffer(col, col_buffers, level + 1));
+        col_buffer.children.emplace_back(assemble_buffer(col, col_buffers, level + 1, stream));
       }
 
       break;
+    case orc::MAP: {
+      std::vector<column_buffer> child_col_buffers;
+      // Get child buffers
+      for (size_t idx = 0; idx < _metadata->get_col_type(orc_col_id).subtypes.size(); idx++) {
+        auto name = (idx == 0) ? "key" : "value";
+        auto col  = _metadata->get_col_type(orc_col_id).subtypes[idx];
+        child_col_buffers.emplace_back(assemble_buffer(col, col_buffers, level + 1, stream));
+        child_col_buffers.back().name = name;
```

Comment:
there's no way to et the name in the constructor?

Reply:
column_buffer currently works on using cudf types, and since we don't support MAP type in cuDF as of now and column buffer is being used by other readers as well, we can't change it.